src/signals/incident-management/containers/IncidentDetail/components/Detail/components/Location/index.test.js

```diff
@@ -1,10 +1,109 @@
-// import React from 'react';
-// import { shallow } from 'enzyme';
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
 
-// import Location from 'index';
+import { getListValueByKey } from 'shared/services/list-helper/list-helper';
+
+import Location from './index';
+
+jest.mock('../../../MapDetail', () => () => <div data-testid="location-map" />);
+jest.mock('shared/services/list-helper/list-helper');
 
 describe('<Location />', () => {
-  it('Expect to have unit tests specified', () => {
-    expect(true).toEqual(true);
+  let props;
+
+  beforeEach(() => {
+    getListValueByKey.mockImplementation(() => 'Centrum');
+
+    props = {
+      incident: {
+        location: {
+          extra_properties: null,
+          geometrie: {
+            type: 'Point',
+            coordinates: [
+              4.892649650573731,
+              52.36918517949316
+            ]
+          },
+          buurt_code: 'A00d',
+          created_by: null,
+          address: {
+            postcode: '1012KP',
+            huisletter: 'A',
+            huisnummer: '123',
+            woonplaats: 'Amsterdam',
+            openbare_ruimte: 'Rokin',
+            huisnummer_toevoeging: 'H'
+          },
+          stadsdeel: 'A',
+          bag_validated: false,
+          address_text: 'Rokin 123-H 1012KP Amsterdam',
+          id: 3372
+        }
+      },
+      stadsdeelList: [
+        {
+          key: 'A',
+          value: 'Centrum'
+        }
+      ],
+      onShowLocation: jest.fn(),
+      onEditLocation: jest.fn()
+    };
+  });
+
+  describe('rendering', () => {
+    it('should render correctly', () => {
+      const { queryByTestId, queryAllByTestId } = render(
+        <Location {...props} />
+      );
+
+      expect(queryByTestId('location-definition')).toHaveTextContent(/^Locatie$/);
+      expect(queryByTestId('location-value-address-stadsdeel')).toHaveTextContent(/^Stadsdeel: Centrum$/);
+      expect(queryByTestId('location-value-address-street')).toHaveTextContent(/^Rokin 123A-H$/);
+      expect(queryByTestId('location-value-address-city')).toHaveTextContent(/^1012KP Amsterdam$/);
+      expect(queryAllByTestId('location-map')).toHaveLength(1);
+    });
+
+    it('should render correctly without huisnummer_toevoeging', () => {
+      props.incident.location.address.huisnummer_toevoeging = undefined;
```

Comment:
What happens when `huisnummer_toevoeging` isn't present at all? So when you do
```jsx
const propsWithoutHuisnummer = Object.assign({}, props);
delete propsWithoutHuisnummer.incident.location.address.huisnummer_toevoeging;
```

Reply:
dat is niet nodig. hij werkt zo goed. de test werkt zoals verwacht.
er wordt in de beforeEach een nieuwe set props gezet. dus er hoeft niet een nieuw object aangemaakt.